### DISCUSSION TITLE
Gitlab hosted release notes not attached on dependency update.

### DISCUSSION THREAD

**Discussion by cykl** [renovatebot/renovate#12522] [Request Help] [answered]
I had like to attach release notes to the merge request created by renovate when an internal package (self hosted Gitlab & Nexus) is updated. This doesn't happen, and debug / trace logs doesn't show any code from lib/workers/pr/changelog/(gitlab)? being executed. My configuration is likely incorrect but I can't figure out was is wrong or how the execution should look like.

My configuration is:

```json
{
      "platform": "gitlab",
      "endpoint": "https://gitlab.xxx.com/api/v4/",

      "packageRules": [
        {
          "datasources": [
            "maven"
          ],
          "registryUrls": [
            "https://nexus.xxx.com/repository/[redacted]"
          ]
        },   
}
```

I can observe that renovate detects that app1 depends on lib1, that a newer package of lib1 is available. But it doesn't seems to even try to fetch lib1/CHANGELOG.md to extract relevant release note. It is also unclear to me how renovate would discover that Maven artifact xxx.com:lib1 is in fact hosted at gitlab.xxx/group/lib1 and then fetch its release note.

Any hint would be welcome.

**Comment by viceice** [comment, marked as answer]
Does the dep in the pr have a link to your source repo? If not, renovate can't find the sourceUrl from the package at your registry.

**Reply by cykl** [reply]
That's it, pom had no scm.url! 

I did grep for scm in manager/maven rather than datasource/maven, found no match, and was struggling to understand how it could work :facepalm:
